QMLComponents/controls/radiobuttonsgroupbase.h

```diff
@@ -38,23 +40,30 @@ class RadioButtonsGroupBase : public JASPControl, public BoundControlBase
 	Json::Value createJson()								const	override;
 	void		bindTo(const Json::Value& value)					override;
 	void		setUp()												override;
+
+	void registerRadioButton(RadioButtonBase* button);
+	void unregisterRadioButton(RadioButtonBase* button);
+	void clickHandler(RadioButtonBase* button);
+	void radioButtonValueChanged(RadioButtonBase* button);
     
 	const QString&	value()									const	{ return _value; }
 
 signals:
 	void valueChanged();
-	void clicked(const QVariant& button);
-
-protected slots:
-	void clickedSlot(const QVariant& button);
+	void clicked();
+	void buttonsChanged();
 
 protected:
 	GENERIC_SET_FUNCTION(Value,		_value,		valueChanged,	QString	)
 
+	RadioButtonBase* checkedButton() { auto it = _valueButtonMap.find(_value); return it == _valueButtonMap.end() ? nullptr : it.value(); }
+	QList<RadioButtonBase*> buttons() { return _valueButtonMap.values(); }
+
 	void _setCheckedButton(RadioButtonBase* button);
-	void _getRadioButtons(QQuickItem* item, QList<RadioButtonBase* >& buttons);
 
-	QMap<QString, RadioButtonBase *>	_buttons;
+
+	QMap<QString, RadioButtonBase *>	_valueButtonMap;
```

Comment:
Im not really sure why you opted to use two maps instead of a single set, but I could very well be missing something